packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/remote-table.service.ts

```diff
@@ -134,71 +124,100 @@ export class RemoteTableService {
     input: RemoteTableInput,
     remoteServer: RemoteServerEntity<RemoteServerType>,
     workspaceId: string,
-    workspaceDataSource: DataSource,
-    dataSourceMetadata: DataSourceEntity,
   ) {
-    const localSchema = dataSourceMetadata.schema;
+    if (!input.schema) {
+      throw new Error('Schema is required for syncing remote table');
+    }
+
+    const currentForeignTableNames =
+      await this.fetchForeignTableNamesWithinWorkspace(workspaceId);
+
+    if (currentForeignTableNames.includes(getRemoteTableName(input.name))) {
+      throw new Error('Remote table already exists');
+    }
 
-    // TODO: Add strong typing for remote table columns. Will be done when we have another use case than Postgres
     const remoteTableColumns = await this.fetchTableColumnsSchema(
       remoteServer,
       input.name,
       input.schema,
     );
 
-    const foreignTableColumns = remoteTableColumns
-      .map((column) => `"${column.column_name}" ${column.data_type}`)
-      .join(', ');
-
     const remoteTableName = getRemoteTableName(input.name);
     const remoteTableLabel = camelToTitleCase(remoteTableName);
 
     // We only support remote tables with an id column for now.
     const remoteTableIdColumn = remoteTableColumns.filter(
-      (column) => column.column_name === 'id',
+      (column) => column.columnName === 'id',
     )?.[0];
 
     if (!remoteTableIdColumn) {
       throw new Error('Remote table must have an id column');
     }
 
-    await workspaceDataSource.query(
-      `CREATE FOREIGN TABLE ${localSchema}."${remoteTableName}" (${foreignTableColumns}) SERVER "${remoteServer.foreignDataWrapperId}" OPTIONS (schema_name '${input.schema}', table_name '${input.name}')`,
+    const dataSourcesMetatada =
```

Comment:
we have: getLastDataSourceMetadataFromWorkspaceIdOrFail that returns only one datasource